packages/astro/src/core/routing/manifest/generator.ts

```diff
@@ -49,13 +55,18 @@ export function getRouteGenerator(
 		})
 		.join('');
 
-	// Unless trailingSlash config is set to 'always', don't automatically append it.
 	let trailing: '/' | '' = '';
 	if (addTrailingSlash === 'always' && segments.length) {
 		trailing = '/';
 	}
 	const toPath = compile(template + trailing);
-	return (params: Record<string, string | number | undefined>): string => {
+	return (params: Record<string, string[] | number | undefined>): string => {
+
+		// Convert `slug` to an array if it's a string
+		if (typeof params.slug === 'string') {
```

Comment:
The array conversion should only happen for parameters which  RoutePart has spread set to true. And it should split the string if it contains a /.